Sync 'TextMetrics' to web specification by changing from 'unrestricted float' to 'double'

Ahmad-S792 opened the pull request:
#### 79655bbc8d3b2056a180e69d44d9678722c93043
<pre>
Sync &apos;TextMetrics&apos; to web specification by changing from &apos;unrestricted float&apos; to &apos;double&apos;

<a href="https://bugs.webkit.org/show_bug.cgi?id=262532">https://bugs.webkit.org/show_bug.cgi?id=262532</a>

Reviewed by Chris Dumez.

This patch aligns WebKit with Web-Specification [1] and address &apos;FIXME&apos; present.

[1] <a href="https://html.spec.whatwg.org/multipage/canvas.html#textmetrics">https://html.spec.whatwg.org/multipage/canvas.html#textmetrics</a>

* Source/WebCore/html/TextMetrics.h:
* Source/WebCore/html/TextMetrics.idl:

Canonical link: <a href="https://commits.webkit.org/268786@main">https://commits.webkit.org/268786@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7f3d3dbbdad1f701eab288a136a2e758b809125a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/20672 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/21090 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21745 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/22566 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/19295 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/20909 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/24344 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/21271 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/20627 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/20894 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20719 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/17975 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/23422 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/17885 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/18781 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/25045 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18964 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/18964 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22991 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/19545 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/16607 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18765 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4960 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/23102 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/19363 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/html/TextMetrics.h:
```diff
@@ -34,55 +34,55 @@ class TextMetrics : public RefCounted<TextMetrics> {
-    float m_hangingBaseline { 0 };
-    float m_alphabeticBaseline { 0 };
-    float m_ideographicBaseline { 0 };
+    double m_width { 0.0 };
```
I don't think it is useful to use `0.0` instead of `0` here.

Ahmad-S792 replied:
Done in next push. :-)

Ahmad-S792 replied:
Just to add, I was matching Blink with 0.0.

cdumez reviewed (approved):
LGTM with nit.

Ahmad-S792 committed:
Sync 'TextMetrics' to web specification by changing from 'unrestricted float' to 'double'

https://bugs.webkit.org/show_bug.cgi?id=262532

Reviewed by Chris Dumez.

This patch aligns WebKit with Web-Specification [1] and address 'FIXME' present.

[1] https://html.spec.whatwg.org/multipage/canvas.html#textmetrics

* Source/WebCore/html/TextMetrics.h:
* Source/WebCore/html/TextMetrics.idl:

Canonical link: https://commits.webkit.org/268786@main
